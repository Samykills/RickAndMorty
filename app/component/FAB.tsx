import React, { Component } from 'react';
import { Animated, Platform, Text } from 'react-native';
import { Icon } from 'react-native-elements';
import { Actions } from 'react-native-router-flux';

const HEADER_MAX_HEIGHT = 1;

export interface Props {
  scroll: any;
}

class FAB extends Component<Props> {
  render() {
    const { scroll } = this.props;

    const fabAnimationY = scroll.interpolate({
      inputRange: [0, HEADER_MAX_HEIGHT],
      outputRange: [100, HEADER_MAX_HEIGHT],
      extrapolate: 'clamp'
    });

    return (
      <Animated.View
        style={{
          position: 'absolute',
          bottom: Platform.OS == 'ios' ? 50 : 30,
          right: 25,
          justifyContent: 'center',
          alignItems: 'center',
          transform: [
            {
              translateY: fabAnimationY
            }
          ]
        }}
      >
        <Icon raised name='search' type='font-awesome' color='#f50' onPress={Actions.searchScreen} />
      </Animated.View>
    );
  }
}
export default FAB;
