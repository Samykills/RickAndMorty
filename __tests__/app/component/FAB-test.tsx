// __tests__/Intro-test.js
import React from 'react';
import { Animated } from 'react-native';
import FAB from '../../../app/component/FAB';
import renderer from 'react-test-renderer';

test('FAB renders correctly', () => {
  const tree = renderer.create(<FAB scroll={new Animated.Value(0)} />).toJSON();
  expect(tree).toMatchSnapshot();
});
