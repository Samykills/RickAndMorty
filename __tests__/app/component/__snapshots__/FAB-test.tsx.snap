// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAB renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "bottom": 50,
      "justifyContent": "center",
      "opacity": 0,
      "position": "absolute",
      "right": 25,
      "transform": Array [
        Object {
          "translateY": 200,
        },
      ],
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 28,
          "height": 52,
          "justifyContent": "center",
          "margin": 7,
          "shadowColor": "rgba(0,0,0, .4)",
          "shadowOffset": Object {
            "height": 1,
            "width": 1,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
          "width": 52,
        }
      }
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
      underlayColor="white"
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#263238",
              "fontSize": 24,
            },
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
        testID="iconIcon"
      >
        ï€‚
      </Text>
    </View>
  </View>
</View>
`;
